### PR TITLE
tezos_interop: integrates fetch_storage

### PR DESCRIPTION
**Pull request by EduardoRFS**
## Depends

- [x] #499

## Problem

After #499 Deku should have an unified tezos bridge, but currently `fetch_storage.js` is not integrated on it.

## Solution

This PR moves `fetch_storage.js` entirely to `tezos_js_bridge.js` and properly integrates it on Tezos_interop.

### CODE REVIEW

**Review by d4hines — APPROVED** [review]
LGTM. Appreciate the better abstractions we're forming around the JS layer.

hooray for better abstractions

**Comment by anchpop** [comment]
@EduardoRFS should I rebase #450 onto this branch? I add a js script for grabbing the validators from the discovery.mligo big map, and I think that should be merged into here?

**Comment by EduardoRFS** [comment]
@anchpop I would say yes

File: src/tezos_interop/tezos_js_bridge.js [hunk 1]
```diff
@@ -124,13 +139,56 @@ const onRequest = async (request) => {
...
+  const Tezos = new TezosToolkit(rpc_node);
+  Tezos.setProvider({ config });
+
+  const block = await client.getBlock(); // fetches the head
+  const contract = await client.getContract(destination, {
+    block: block.hash,
+  });
+  const storage = contract.script.storage;
+  /* To make sure the storage state is finalized, we query the last
+     block and any one of it's operation, and wait till it receives
```
Comment (gsebil08) [inline src/tezos_interop/tezos_js_bridge.js:155]:
```suggestion
     block and any one of its operation, and wait till it receives
```

File: src/tezos_interop/tezos_js_bridge.js [hunk 2]
```diff
@@ -124,13 +139,56 @@ const onRequest = async (request) => {
...
+  const { rpc_node, confirmation, destination } = request;
+  const client = new RpcClient(rpc_node);
+  const Tezos = new TezosToolkit(rpc_node);
+  Tezos.setProvider({ config });
+
+  const block = await client.getBlock(); // fetches the head
+  const contract = await client.getContract(destination, {
+    block: block.hash,
+  });
+  const storage = contract.script.storage;
```
Comment (gsebil08) [inline src/tezos_interop/tezos_js_bridge.js:153]: Why extracting it so long before needing it? I expected to find some transformation on it. I think this line should be right before the `return` one.

File: src/tezos_interop/tezos_js_bridge.js [hunk 3]
```diff
@@ -104,7 +119,7 @@ const config = {
 };
 
 /** @param {TransactionRequest} request */
-const onRequest = async (request) => {
+const onTransactionRequest = async (request) => {
```
Comment (gsebil08) [inline src/tezos_interop/tezos_js_bridge.js:122]: Clever renaming, thanks.

**Review by aguillon — APPROVED** [review]

File: src/tezos_interop/tezos_bridge.ml [hunk 4]
```diff
@@ -50,6 +53,42 @@ module Transaction = struct
     | _ -> Error "invalid status"
 end
 
+module Storage = struct
+  type kind = Storage
+  let kind_to_yojson Storage = `String "storage"
```
Comment (gsebil08) [inline src/tezos_interop/tezos_bridge.ml:58]: Noob question: would it be the same to do:
```ocaml
type kind = Storage [@name storage]
  [@@deriving to_yojson]
```

**Review by gsebil08 — APPROVED** [review]
Newbie questions, but LGTM.

File: src/tezos_interop/tezos_bridge.ml (hunk 4 above)
Comment (EduardoRFS) [reply src/tezos_interop/tezos_bridge.ml:58]: @gsebil08 nope, also tried that, TLDR variants are encoded as `[header, ...values]` so it endup as `["storage"]`

File: src/tezos_interop/tezos_js_bridge.js (hunk 2 above)
Comment (EduardoRFS) [reply src/tezos_interop/tezos_js_bridge.js:153]: I just kept the code in the same order, this is quite fragile and I didn't want to break it.

**Review by anchpop — APPROVED** [review]

**Review by aguillon — APPROVED** [review]

**Pull request merged**